src/boxes.js

```diff
@@ -46,124 +50,84 @@ SVG.Box = SVG.invent({
-      bbox.x = xMin
-      bbox.width = xMax-xMin
-      bbox.y = yMin
-      bbox.height = yMax-yMin
+      return new SVG.Box(
+        xMin, yMin,
+        xMax-xMin,
+        yMax-yMin
+      )
+    }
+    
+  , addOffset: function() {
+      // offset by window scroll position, because getBoundingClientRect changes when window is scrolled
+      this.x += window.pageXOffset
+      this.y += window.pageYOffset
+      return this
+    }
+  , toString: function() {
+      return this.x + ' ' + this.y + ' ' + this.width + ' ' + this.height
+    }
+  , morph: function(x, y, width, height){
+      this.destination = new SVG.Box(x, y, width, height)
+      return this
+    }
+
+  , at: function(pos) {
 
-      fullBox(bbox)
+      if(!this.destination) return this
+
+      return new SVG.Box(
+          this.x + (this.destination.x - this.x) * pos
+        , this.y + (this.destination.y - this.y) * pos
+        , this.width + (this.destination.width - this.width) * pos
+        , this.height + (this.destination.height - this.height) * pos
+      )
 
-      return bbox
     }
   }
-})
-
-SVG.BBox = SVG.invent({
-  // Initialize
-  create: function(element) {
-    SVG.Box.apply(this, [].slice.call(arguments))
+  
+    // Define Parent
+, parent: SVG.Element
 
-    // get values if element is given
-    if (element instanceof SVG.Element) {
+  // Constructor
+, construct: {
+    // Get bounding box
+    bbox: function() {
       var box
-
-      // yes this is ugly, but Firefox can be a bitch when it comes to elements that are not yet rendered
+    
       try {
-
-        if (!document.documentElement.contains){
-          // This is IE - it does not support contains() for top-level SVGs
-          var topParent = element.node;
-          while (topParent.parentNode){
-            topParent = topParent.parentNode;
-          }
-          if (topParent != document) throw new Exception('Element not in the dom')
-        } else {      
-          // the element is NOT in the dom, throw error
-          if(!document.documentElement.contains(element.node)) throw new Exception('Element not in the dom')
-        }
-
         // find native bbox
-        box = element.node.getBBox()
+        box = this.node.getBBox()
+        
+        if(isNulledBox(box) && !domContains(this.node)) {
+          throw new Exception('Element not in the dom')
+        }
       } catch(e) {
-        if(element instanceof SVG.Shape){
-          var clone = element.clone(SVG.parser.draw).show()
+        try {
+          var clone = this.clone(SVG.parser.draw).show()
           box = clone.bbox()
           clone.remove()
-        }else{
-          box = {
-            x:      element.node.clientLeft
-          , y:      element.node.clientTop
-          , width:  element.node.clientWidth
-          , height: element.node.clientHeight
-          }
+        } catch(e) { 
+          console.warn('Getting a bounding box of this element is not possible')
```

Comment:
I guess we dont need that try catch here because catch wont be ever triggered.
It will be an infinitive loop instead because bbox is called in the try block.
Maybe I can rewrite it to make it better